src/observers/stores/opentelemetry.py

```diff
@@ -0,0 +1,109 @@
+# stdlib features
+from dataclasses import dataclass
+from typing import Optional
+
+# Observers internal interfaces
+from observers.observers.base import Record
+from observers.stores.base import Store
+
+# Actual dependencies
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider, Tracer, Span
+from opentelemetry.sdk.trace.export import (
+    BatchSpanProcessor,
+)
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+
+
+def flatten_dict(d, prefix=""):
+    """Flatten a python dictionary, turning nested keys into dotted keys"""
+    flat = {}
+    for k, v in d.items():
+        if v:
+            if type(v) is dict:
+                if prefix:
+                    flat.extend(flatten_dict(v, f"{prefix}.{k}"))
+            else:
+                if prefix:
+                    flat[(f"{prefix}.{k}")] = v
+                else:
+                    flat[k] = v
+
+
+@dataclass
+class OpenTelemetryStore(Store):
+    """
+    OpenTelemetry Store
+    """
+
+    # These are here largely to ease future refactors/conform to
+    # the style of the other stores. They have defaults set in the constructor,
+    # but, set here as well.
+    tracer: Optional[Tracer] = None
+    root_span: Optional[Span] = None
+    connected: bool = False
+
+    def __post_init__(self):
+        # if we weren't called by the `connect` constructor, we still need to
+        # `connect`, and the example construction uses the ObjectName() syntax
+        if not self.connected:
+            self.tracer, self.connected = OpenTelemetryStore._connect()
+        # if we initialize a span here, then all subsequent 'add's can be
+        # added to a continuous trace
+        with self.tracer.start_as_current_span("observers.init") as span:
+            span.set_attribute("connected", True)
+            self.root_span = span
+
+    def add(self, record: Record):
+        """Add a new record to the store"""
+        with trace.use_span(self.root_span):
+            with self.tracer.start_as_current_span("observers.add") as span:
+                # Split out to be easily edited if the record api changes
+                event_fields = [
```

Comment:
in the land of OTel these things would normally be prefixed with some kind of namespace, maybe HF-related. But looking at other code in this repo and these are the names used, so it's probably alright.

Reply:
tacking huggingface on feels legit. maybe i'll pull that out into a lil' global such that if conventions change maint is easy